aten/src/ATen/native/Math.h

```diff
@@ -324,15 +329,20 @@ static inline double calc_digamma(double x) {
  * See note [3-Clause BSD License for the Cephes Math Library].
  */
 static inline float calc_digamma(float x) {
+  // See [C++ Standard Reference: Gamma Function]
   static float PSI_10 = 2.25175258906672110764f;
   if (x == 0) {
-    return INFINITY;
+    // As per C++ standard for gamma related functions and SciPy,
+    // If the argument is ±0, ±∞ is returned
+    return std::copysign(INFINITY, -x);
   }
 
-  int x_is_integer = x == floorf(x);
+  int x_is_integer = x == truncf(x);
```

Comment:
Internal linter correctly points out that `x_is_integer` should be a bool, not int, in all the kernels. Should be an easy change.

Reply:
Ah. Right. Done!